### PR TITLE
godsflaw-20211115: improves deploy scripts

### PR DESCRIPTION
**Pull request by godsflaw**
- changes kovan to goerli
- adds some flatten sources
- adds deploy code for `UniswapV3Callee`
- ~adds deploy code for `UniV3Callee`~
- adds deploy code for `WstETHCurveUniv3Callee`

```haskell
Running 4 tests for src/test/OasisDexCallee.t.sol:CalleeOtcDaiTest
[PASS] test_rounding_error() (gas: 2702244)
[PASS] test_flash_take_profit() (gas: 3014054)
[PASS] test_flash_take_profit_thin_orderbook() (gas: 2771405)
[PASS] test_flash_take_no_profit() (gas: 2691056)

Running 1 tests for src/test/OasisDexCalleeCharter.t.sol:CalleeOtcDaiTest
[PASS] test_flash_take_profit() (gas: 3166624)

Running 21 tests for src/test/Simulation.t.sol:SimulationTests
[PASS] testFrobMax() (gas: 343291)
[PASS] testTakeLpDaiEthProfit() (gas: 8902803)
[PASS] testGetWeth() (gas: 36273)
[PASS] testTakeLinkV2Profit() (gas: 4435263)
[PASS] testGetLpDaiEth() (gas: 302575)
[PASS] testBarkLpDaiEth() (gas: 883498)
[PASS] testBurnLpDaiEth() (gas: 423682)
[PASS] testDrip() (gas: 54324)
[PASS] testTakeLinkV2NoProfit() (gas: 2985304)
[PASS] testGetLpDaiEthPrice() (gas: 5551)
[PASS] testTakeLpDaiEthNoProfit() (gas: 4736609)
[PASS] testGetDai() (gas: 156462)
[PASS] testGetLinkPrice() (gas: 5684)
[PASS] testGetEthPrice() (gas: 5570)
[PASS] testTakeLinkV3NoProfit() (gas: 3068444)
[PASS] testBarkLink() (gas: 722623)
[PASS] testSwapLinkDai() (gas: 319534)
[PASS] testJoinLink() (gas: 240864)
[PASS] testTakeLinkV3Profit() (gas: 4531830)
[PASS] testJoinLpDaiEth() (gas: 380897)
[PASS] testGetLink() (gas: 165913)

Running 5 tests for src/test/UniswapV2Callee.t.sol:UniswapV2CalleeDaiTest
[PASS] test_complex_path() (gas: 2857662)
[PASS] test_rounding_error() (gas: 2857394)
[PASS] test_flash_take_profit() (gas: 3174834)
[PASS] test_flash_take_profit_thin_orderbook() (gas: 2926533)
[PASS] test_flash_take_no_profit() (gas: 2846184)

Running 1 tests for src/test/UniswapV2CalleeCharter.t.sol:UniswapV2CalleeDaiTest
[PASS] test_flash_take_profit() (gas: 3327236)

Running 7 tests for src/test/WstETHCurveUniv3Callee.t.sol:CurveCalleeTest
[PASS] test_profit() (gas: 1187325)
[PASS] test_baseline() (gas: 1190352)
[PASS] test_poolFee() (gas: 1183214)
[PASS] testFail_badPoolFee() (gas: 1014422)
[PASS] testFail_maxPrice() (gas: 697567)
[PASS] test_bigAmt() (gas: 1202605)
[PASS] test_maxPrice() (gas: 1217455)
```

### CODE REVIEW

**Comment by julienmartinlevrai** [comment]
@godsflaw : `UniV3Callee` does not exist anymore. During the UniV3 PR review process, @brianmcmichael suggested to rename it to `UniswapV3Callee` in order to respect convention. Is there still a `UniV3Callee` somewhere in the code ? If there is, it is my mistake and should be corrected.

**Comment by godsflaw** [comment]
> @godsflaw : `UniV3Callee` does not exist anymore. During the UniV3 PR review process, @brianmcmichael suggested to rename it to `UniswapV3Callee` in order to respect convention. Is there still a `UniV3Callee` somewhere in the code ? If there is, it is my mistake and should be corrected.

I did see the discussion, but yes, perhaps we missed this in the curve PR:
https://github.com/makerdao/exchange-callees/blob/master/src/UniV3Callee.sol#L56

File: scripts/deploy-mainnet.sh [hunk 1]
```diff
@@ -8,5 +8,22 @@ UniswapV2LpTokenCalleeDai=$(dapp create UniswapV2LpTokenCalleeDai \
...
+  0x9759A6Ac90977b93B58547b4A71c78317f391A28)
+
+UniV3Callee=$(dapp create UniV3Callee \
+  0xE592427A0AEce92De3Edee1F18E0157C05861564 \
+  0x9759A6Ac90977b93B58547b4A71c78317f391A28)
+
+WstETHCurveUniv3Callee=$(dapp create WstETHCurveUniv3Callee \
+  0xDC24316b9AE028F1497c275EB9192a3Ea0f67022 \
+  ${UniV3Callee} \
+  0x6a60b7070befb2bfc964F646efDF70388320f4E0 \
```
Comment (iamchrissmith) [inline scripts/deploy-mainnet.sh:22]: This is the goerli DaiJoin address, this should be: `0x9759A6Ac90977b93B58547b4A71c78317f391A28`

File: scripts/deploy-mainnet.sh [hunk 2]
```diff
@@ -8,5 +8,22 @@ UniswapV2LpTokenCalleeDai=$(dapp create UniswapV2LpTokenCalleeDai \
...
+  0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6)
```
Comment (iamchrissmith) [inline scripts/deploy-mainnet.sh:23]: also the goerli address, mainnet WETH is (I think): `0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2`

**Review by iamchrissmith — CHANGES_REQUESTED** [review]
@godsflaw looks like some goerli addresses got into the mainnet script. Please double check i found the right ones. Also does this signify some deeper issue with the tests if things passed? Or did you run them just against goerli?

File: scripts/deploy-mainnet.sh (hunk 1 above)
Comment (godsflaw) [reply scripts/deploy-mainnet.sh:22]: This is why I can't code in meetings.

**Comment by julienmartinlevrai** [comment]
> > @godsflaw : `UniV3Callee` does not exist anymore. During the UniV3 PR review process, @brianmcmichael suggested to rename it to `UniswapV3Callee` in order to respect convention. Is there still a `UniV3Callee` somewhere in the code ? If there is, it is my mistake and should be corrected.
> 
> I did see the discussion, but yes, perhaps we missed this in the curve PR: https://github.com/makerdao/exchange-callees/blob/master/src/UniV3Callee.sol#L56

Probably a merge conflict I didn’t noticed. I just diffed the files and they are the same, so I am going to delete the file and update the déploy scripts and push to this branch, hope you don’t mind.

**Review by iamchrissmith — APPROVED** [review]
changes look good to me and ran tests locally ✅

**Review by julienmartinlevrai — CHANGES_REQUESTED** [review]
Checked:
* Mainnet UniV3 Router matches https://github.com/Uniswap/v3-periphery/blob/main/deploys.md
* Mainnet daiJoin matches https://changelog.makerdao.com/releases/mainnet/1.9.9/contracts.json
* Mainnet Curve Pool matches https://curve.readthedocs.io/ref-addresses.html
* Mainnet WETH matches https://changelog.makerdao.com/releases/mainnet/1.9.9/contracts.json
* Görli UniV2 Router matches https://docs.uniswap.org/protocol/V2/reference/smart-contracts/router-02
* Görli daiJoin matches https://changelog.makerdao.com/releases/goerli/1.9.9/contracts.json
* Görli UniV3 Router matches https://github.com/Uniswap/v3-periphery/blob/main/deploys.md
* Görli WETH matches https://changelog.makerdao.com/releases/goerli/1.9.9/contracts.json

Finding:
* Instead of using the address for the UniV3 Router, the `WstEthCurveUniv3Callee` contract was using the address for the `UniswapV3Callee` contract. See suggested changes below.

**Review by julienmartinlevrai — APPROVED** [review]
LGTM

**Comment by godsflaw** [comment]
```haskell
Running 4 tests for src/test/OasisDexCallee.t.sol:CalleeOtcDaiTest
[PASS] test_rounding_error() (gas: 2702244)
[PASS] test_flash_take_profit() (gas: 3014054)
[PASS] test_flash_take_profit_thin_orderbook() (gas: 2771405)
[PASS] test_flash_take_no_profit() (gas: 2691056)

Running 1 tests for src/test/OasisDexCalleeCharter.t.sol:CalleeOtcDaiTest
[PASS] test_flash_take_profit() (gas: 3166624)

Running 21 tests for src/test/Simulation.t.sol:SimulationTests
[PASS] testFrobMax() (gas: 343291)
[PASS] testTakeLpDaiEthProfit() (gas: 8902583)
[PASS] testGetWeth() (gas: 36273)
[PASS] testTakeLinkV2Profit() (gas: 4435445)
[PASS] testGetLpDaiEth() (gas: 302575)
[PASS] testBarkLpDaiEth() (gas: 883278)
[PASS] testBurnLpDaiEth() (gas: 423682)
[PASS] testDrip() (gas: 54506)
[PASS] testTakeLinkV2NoProfit() (gas: 2985486)
[PASS] testGetLpDaiEthPrice() (gas: 5551)
[PASS] testTakeLpDaiEthNoProfit() (gas: 4736389)
[PASS] testGetDai() (gas: 156462)
[PASS] testGetLinkPrice() (gas: 5684)
[PASS] testGetEthPrice() (gas: 5570)
[PASS] testTakeLinkV3NoProfit() (gas: 3068626)
[PASS] testBarkLink() (gas: 722805)
[PASS] testSwapLinkDai() (gas: 319534)
[PASS] testJoinLink() (gas: 240864)
[PASS] testTakeLinkV3Profit() (gas: 4532012)
[PASS] testJoinLpDaiEth() (gas: 380897)
[PASS] testGetLink() (gas: 165913)

Running 5 tests for src/test/UniswapV2Callee.t.sol:UniswapV2CalleeDaiTest
[PASS] test_complex_path() (gas: 2857662)
[PASS] test_rounding_error() (gas: 2857394)
[PASS] test_flash_take_profit() (gas: 3174834)
[PASS] test_flash_take_profit_thin_orderbook() (gas: 2926533)
[PASS] test_flash_take_no_profit() (gas: 2846184)

Running 1 tests for src/test/UniswapV2CalleeCharter.t.sol:UniswapV2CalleeDaiTest
[PASS] test_flash_take_profit() (gas: 3327236)

Running 7 tests for src/test/WstETHCurveUniv3Callee.t.sol:CurveCalleeTest
[PASS] test_profit() (gas: 1187105)
[PASS] test_baseline() (gas: 1190132)
[PASS] test_poolFee() (gas: 1182994)
[PASS] testFail_badPoolFee() (gas: 1014202)
[PASS] testFail_maxPrice() (gas: 697347)
[PASS] test_bigAmt() (gas: 1202385)
[PASS] test_maxPrice() (gas: 1217235)
```

**Pull request merged**